lib/algos/list/delimiterMaster.js

```diff
@@ -95,12 +95,29 @@ class DelimiterMaster extends Delimiter {
              *   NextMarker to the common prefix instead of the whole key
              *   value. (TODO: remove this test once ZENKO-1048 is fixed)
              *   */
-            if (key === this.prvKey || key === this[this.nextContinueMarker] ||
-                (this.delimiter &&
-                key.startsWith(this[this.nextContinueMarker]))) {
+            if (key === this.prvKey || key === this[this.nextContinueMarker]) {
                 /* master version already filtered */
                 return FILTER_SKIP;
             }
+
+            if (this.delimiter && key.startsWith(this[this.nextContinueMarker])) {
+                // If this[continueMarker] ends with the delimiter then we have to treat it differently.
+                const candidate = key.split(this.delimiter);
```

Comment:
I don't think splitting the key is really needed for what we want to check, essentially:
- if the key ends with a delimiter character
- if the key contains no delimiter at all in its whole length (not sure why this check)

I'd rather avoid the split call and just do the same kind of checks on the whole key to avoid unnecessary processing and simplify the code.